testnet/stacks-node/src/tests/epoch_22.rs

```diff
@@ -0,0 +1,1168 @@
+    let spender_2_account = get_account(&http_origin, &spender_2_addr);
+
+    info!("spender_1_account = {:?}", spender_1_account);
+    info!("spender_2_account = {:?}", spender_1_account);
+
+    assert_eq!(
+        spender_1_account.balance,
+        spender_1_initial_balance as u128 - (2 * tx_fee as u128),
+        "Spender 1 should be unlocked"
+    );
+    assert_eq!(spender_1_account.locked, 0, "Spender 1 should be unlocked");
+    assert_eq!(
+        spender_1_account.nonce, 2,
+        "Spender 1 should have two accepted transactions"
+    );
+
+    assert_eq!(
+        spender_2_account.balance,
+        spender_2_initial_balance as u128 - (1 * tx_fee as u128),
+        "Spender 2 should be unlocked"
+    );
+    assert_eq!(spender_2_account.locked, 0, "Spender 2 should be unlocked");
+    assert_eq!(
+        spender_2_account.nonce, 1,
+        "Spender 2 should have two accepted transactions"
+    );
+
+    // perform a transfer
+    let tx = make_stacks_transfer(&spender_sk, 2, tx_fee, &spender_3_addr, 1_000_000);
+
+    info!("Submit stack transfer tx to {:?}", &http_origin);
+    submit_tx(&http_origin, &tx);
+
+    // this wakes up the node to mine the transaction
+    next_block_and_wait(&mut &mut btc_regtest_controller, &blocks_processed);
+    // this block selects the previously mined block
+    next_block_and_wait(&mut &mut btc_regtest_controller, &blocks_processed);
+
+    let spender_1_account = get_account(&http_origin, &spender_addr);
+    let spender_2_account = get_account(&http_origin, &spender_2_addr);
+    let spender_3_account = get_account(&http_origin, &spender_3_addr);
+
+    info!("spender_1_account = {:?}", spender_1_account);
+    info!("spender_2_account = {:?}", spender_1_account);
+
+    assert_eq!(
+        spender_3_account.balance, 1_000_000,
+        "Recipient account should have funds"
+    );
+    assert_eq!(
+        spender_3_account.locked, 0,
+        "Burn account should be unlocked"
+    );
+    assert_eq!(
+        spender_3_account.nonce, 0,
+        "Burn should have no accepted transactions"
+    );
+
+    assert_eq!(
+        spender_1_account.balance,
+        spender_1_initial_balance as u128 - (3 * tx_fee as u128) - 1_000_000,
+        "Spender 1 should be unlocked"
+    );
+    assert_eq!(spender_1_account.locked, 0, "Spender 1 should be unlocked");
+    assert_eq!(
+        spender_1_account.nonce, 3,
+        "Spender 1 should have three accepted transactions"
+    );
+
+    assert_eq!(
+        spender_2_account.balance,
+        spender_2_initial_balance as u128 - (1 * tx_fee as u128),
+        "Spender 2 should be unlocked"
+    );
+    assert_eq!(spender_2_account.locked, 0, "Spender 2 should be unlocked");
+    assert_eq!(
+        spender_2_account.nonce, 1,
+        "Spender 2 should have two accepted transactions"
+    );
+
+    // finish the cycle after the 2.2 transition,
+    //  and mine two more cycles
```

Comment:
"and mine two more cycles" - not sure if this comment is accurate

Reply:
You can see in the expected reward sets that this is checked to be the case -- there's two 2.1 cycles and then twp 2.2 cycles.